opensrp-app/src/main/java/org/smartregister/repository/dao/TaskDaoImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.smartregister.repository.dao;
+
+import android.content.Intent;
+import android.support.v4.content.LocalBroadcastManager;
+
+import com.ibm.fhir.model.resource.QuestionnaireResponse;
+import com.ibm.fhir.model.resource.Task;
+import com.ibm.fhir.path.FHIRPathElementNode;
+
+import org.smartregister.CoreLibrary;
+import org.smartregister.converters.TaskConverter;
+import org.smartregister.pathevaluator.PathEvaluatorLibrary;
+import org.smartregister.pathevaluator.dao.TaskDao;
+import org.smartregister.repository.TaskNotesRepository;
+import org.smartregister.repository.TaskRepository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.smartregister.AllConstants.INTENT_KEY.TASK_GENERATED;
+import static org.smartregister.AllConstants.INTENT_KEY.TASK_GENERATED_EVENT;
+
+/**
+ * Created by samuelgithengi on 9/3/20.
+ */
+public class TaskDaoImpl extends TaskRepository implements TaskDao {
+    public TaskDaoImpl(TaskNotesRepository taskNotesRepository) {
+        super(taskNotesRepository);
+    }
+
+    @Override
+    public List<Task> findTasksForEntity(String id, String planIdentifier) {
+        return getTasksByPlanAndEntity(planIdentifier, id)
+                .stream()
+                .map(TaskConverter::convertTasktoFihrResource)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void saveTask(org.smartregister.domain.Task task, QuestionnaireResponse questionnaireResponse) {
+        if (questionnaireResponse != null) {
+            FHIRPathElementNode structure = PathEvaluatorLibrary.getInstance()
+                    .evaluateElementExpression(questionnaireResponse,
+                            "$this.item.where(url='details' and linkId='location_id').answer");
```

Comment:
Could we make this a constant with a descriptive name.

Reply:
You mean constants for the whole expression